pkg/apis/storage/validation/validation.go

```diff
@@ -266,3 +278,180 @@ func validateAllowedTopologies(topologies []api.TopologySelectorTerm, fldPath *f
 
 	return allErrs
 }
+
+// ValidateCSINodeInfo validates a CSINodeInfo.
+func ValidateCSINodeInfo(csiNodeInfo *storage.CSINodeInfo) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&csiNodeInfo.ObjectMeta, false, apivalidation.ValidateClassName, field.NewPath("metadata"))
```

Comment:
You should probably define your own name validator rather than piggy back on Class, or at least make it REALLY clear why Class is the right abstract validator.  E.g. Class is defined as DNS Subdomain.  why is this not saying DNS Subdomain?  Or NodeName, if this is 1:1 with node?

Reply:
You are right.  Will change to ValidateNodeName as this should be the same as K8S node name.